test/controllers/trips_controller_test.rb

```diff
@@ -1,23 +1,126 @@
 require "test_helper"
 
-describe TripsController do
+describe TripsController do  
+  let(:driver) {
+    Driver.create name: "Test Smith", vin: "ADADEVELOPERSACAD",
+    available: "true"
+  }
+  let(:passenger) {
+    Passenger.create name: "Test", phone_num: '123'
+  }
+  let(:trip) {
+    new_trip = Trip.create driver_id: driver.id, passenger_id: passenger.id,
+      date: "2020-05-11", cost: "123"
+    new_trip
+  }
+
+  before (:each) {
+    Trip.delete_all
+    Passenger.delete_all
+    Driver.delete_all
+  }
   describe "show" do
-    # Your tests go here
+    it "responds with success when showing an existing valid trip" do
+      # Arrange
+      driver.save
+      passenger.save
+      trip.save
+      # Act
+      get trip_path(trip.id)
+      # Assert
+      must_respond_with :success
+    end
+
+    it "responds with redirect with an invalid trip id" do
+      # Act
+      get trip_path(-1)
+      # Assert
+      must_respond_with :redirect
+    end
   end
 
   describe "create" do
-    # Your tests go here
+    it "can create a new trip with valid information accurately, and redirect" do
+      # Arrange
+      # Set up the form data
+      trip_hash = {
+        trip: {
+        driver_id: driver.id, passenger_id: passenger.id,
+        date: "2020-05-11", cost: "123"
+        },
+      }
+      # Act-Assert
+      # Ensure that there is a change of 1 in Driver.count
+      expect {
+        post trips_path, params: trip_hash
```

Comment:
When you create a trip you're not submitting a form, so this isn't accurate.  Instead you use a link, so there will be no trip_hash